pallets/credit/src/lib.rs

```diff
@@ -0,0 +1,293 @@
+            let sender = ensure_signed(origin)?;
+            let res = Self::initialize_credit(sender.clone(), credit);
+            if res == true{
+                Self::deposit_event(RawEvent::CreditInitSuccess(sender, credit));
+                Ok(())
+            }else{
+                Self::deposit_event(RawEvent::CreditInitFailed(sender, credit));
+                Err(dispatch::DispatchError::Other(
+                    "CreditInitFailed",
+                ))
+            }
+
+        }
+
+        // clear credit score
+        #[weight = 10_000]
+        pub fn kill_credit_extrinsic(origin) -> dispatch::DispatchResult{
+            let sender = ensure_signed(origin)?;
+            let res = Self::kill_credit(sender.clone());
+            if res == true {
+                Self::deposit_event(RawEvent::KillCreditSuccess(sender));
+                Ok(())
+            }else{
+                Self::deposit_event(RawEvent::KillCreditFailed(sender));
+                Err(dispatch::DispatchError::Other(
+                    "KillCreditFailed",
+                ))
+            }
+
+
+        }
+
+        // Anything that needs to be done at the end of the block.
+        fn on_finalize(_n: T::BlockNumber) {
+            // We update credit score of account per era
+            // Notice: the new_micropayment_size_in_block is block level aggregation, here we need
+            // to aggregate total payment and number of clients first before pass it into update_credit's input
+            if _n % T::BlocksPerEra::get() == T::BlockNumber::default() {
+                // update credit score per era
+                let from = _n.saturating_sub(T::BlocksPerEra::get());
+                let to = _n.saturating_sub(T::BlockNumber::from(1));
+                log!(info, "micropayment_statistics block number from {:?} - to {:?}", from, to);
+                let micropayment_vec = pallet_micropayment::Module::<T>::micropayment_statistics(from, to);
+                Self::update_credit(micropayment_vec);
+
+                // attenuate credit score per era
+                Self::attenuate_credit(_n);
+            }
+        }
+    }
+}
+
+impl<T: Trait> Module<T> {
+    /// init credit score
+    pub fn initialize_credit(account_id: T::AccountId, score: u64) -> bool {
+        // in general, a user start from initial score = 0; with coupon, a user can
+        // start from initial score at most CREDIT_INIT_SCORE
+        // TODO: i.e. add coupon verification for non-zero init credit score
+        if !UserCredit::<T>::contains_key(account_id.clone()) && score < CREDIT_INIT_SCORE {
+            UserCredit::<T>::insert(account_id.clone(), 0);
+            true
+        } else {
+            false
+        }
+    }
+
+    /// update credit score per era using micropayment vec
+    pub fn update_credit(micropayment_vec: Vec<(T::AccountId, BalanceOf<T>, u32)>) {
+        for (server_id, balance, size) in micropayment_vec {
+            if size >= 1 {
```

Comment:
we only update the credit if the size is greater than 1, i.e.  if size > 1

Reply:
We will change to `if size > 1`

Reply:
got it. thanks!